backend/corpora/common/authorizer.py

```diff
@@ -1,54 +1,54 @@
-import os
 from functools import lru_cache
-
 import requests
 from chalice import UnauthorizedError
 from jose import jwt, JWTError
-
-USERINFO_ENDPOINT = "https://czi-single-cell.auth0.com/userinfo"
-
-AUTH0_DOMAIN = "czi-single-cell.auth0.com"
-API_AUDIENCE = f"https://api.{os.environ['DEPLOYMENT_STAGE']}.corpora.cziscience.com"
-ALGORITHMS = ["RS256"]
+from .corpora_config import CorporaAuthConfig
 
 
-def assert_authorized(headers: dict) -> dict:
+def assert_authorized_token(token: str) -> dict:
     """
     Determines if the Access Token is valid and return the decoded token. Userinfo is added to the token if it exists.
-    :param headers: The http headers from the request.
+    :param token: The token
     :return: The decoded access token and userinfo.
     """
-
-    token = get_token_auth_header(headers)
     try:
         unverified_header = jwt.get_unverified_header(token)
     except JWTError:
         raise UnauthorizedError(msg="Unable to parse authentication token.")
-    public_keys = get_public_keys(AUTH0_DOMAIN)
+    auth_config = CorporaAuthConfig()
+    auth0_domain = auth_config.api_base_url
+    audience = auth_config.audience
+    public_keys = get_public_keys(auth0_domain)
     public_key = public_keys.get(unverified_header["kid"])
     if public_key:
+        algorithms = ["RS256"]
         try:
-            payload = jwt.decode(
-                token, public_key, algorithms=ALGORITHMS, audience=API_AUDIENCE, issuer=f"https://{AUTH0_DOMAIN}/"
-            )
+            if not auth0_domain.endswith("/"):
```

Comment:
don't we produce this value? why would we be missing the `/`?

Reply:
Just to be less picky about the auth0_domain.   Otherwise when setting the value in the secret you'd have to know that the code that used it expected an extra /.  So, this is just to allow either.   True, we control the value.